### PR TITLE
Shutdown instead of terminate process on Windows

### PR DESCRIPTION
**Pull request by darstahl**
**- What I did**

This is the first step for solving https://github.com/docker/docker/issues/25982. Platform changes are also needed.

**- How I did it**

Changed the behaviour of signaling a Windows container (docker stop) from terminating the init process to shutting down the container.

**- How to verify it**

Manual verification.

Signed-off-by: Darren Stahl <darst@microsoft.com>

### CODE REVIEW

File: libcontainerd/client_windows.go [hunk 1]
```diff
@@ -397,9 +397,9 @@ func (clnt *client) Signal(containerID string, sig int) error {
 		}
 	} else {
 		// Terminate Process
```
Comment (jhowardmsft) [inline libcontainerd/client_windows.go]: Comment needs updating

**Review by lowenna — CHANGES_REQUESTED** [review]
Nit over comment. This is what I expected would need to be done. I'll talk offline about the platform change, but I guess that means it's an RS2 fix only.

**Comment by darrenstahlmsft** [comment]
Updated to do an async shutdown rather than using the synchronous wrapper, since signal should not block for the container to exit.

**Comment by jhowardmsft** [comment]
LGTM

**Review by lowenna — APPROVED** [review]

**Review by johnstep — APPROVED** [review]
LGTM

**Review by vdemeester — APPROVED** [review]
LGTM :frog:

**Comment by thaJeztah** [comment]
@jhowardmsft wondering; for a future improvement should we use `STOPSIGNAL` to switch between these?

**Comment by jhowardmsft** [comment]
Ah no, I don't think that makes sense.

**Comment by godefroi** [comment]
@jhowardmsft Are the "platform changes" going to have to wait until fall 2017 for the release of RS3, or are we going to get a patch/hotfix sometime before that?

**Comment by jhowardmsft** [comment]
The fix is still being worked on. I can't realistically comment on when it will be available, or whether it will be back-ported at this time.

**Comment by rgl** [comment]
@darrenstahlmsft @jhowardmsft @johnstep can you comment on the related moby https://github.com/moby/moby/issues/25982 issue?

**Pull request merged**